src/gui/plugins/visualize_contacts/VisualizeContacts.cc

```diff
@@ -0,0 +1,393 @@
+          this->dataPtr->worldName = _name->Data();
+          return false;
+        });
+    }
+
+    // Enable collisions
+    this->dataPtr->CreateCollisionData(_ecm);
+    this->dataPtr->initialized = true;
+  }
+
+  {
+    std::lock_guard<std::mutex> lock(this->dataPtr->serviceMutex);
+    if (this->dataPtr->checkboxPrevState && !this->dataPtr->checkboxState)
+    {
+      // Remove the markers
+      this->dataPtr->positionMarkerMsg.set_action(
+        ignition::msgs::Marker::DELETE_ALL);
+      this->dataPtr->forceMarkerMsg.set_action(
+        ignition::msgs::Marker::DELETE_ALL);
+
+      igndbg << "Removing markers..." << std::endl;
+      this->dataPtr->node.Request(
+        "/marker", this->dataPtr->positionMarkerMsg);
+      this->dataPtr->node.Request(
+        "/marker", this->dataPtr->forceMarkerMsg);
+
+      // Change action in case checkbox is checked again
+      this->dataPtr->positionMarkerMsg.set_action(
+        ignition::msgs::Marker::ADD_MODIFY);
+      this->dataPtr->forceMarkerMsg.set_action(
+        ignition::msgs::Marker::ADD_MODIFY);
+    }
+
+    this->dataPtr->checkboxPrevState = this->dataPtr->checkboxState;
+    if (!this->dataPtr->checkboxState)
+      return;
+  }
+
+  // Only publish markers if enough time has passed
+  auto timeDiff =
+    std::chrono::duration_cast<std::chrono::milliseconds>(_info.simTime -
+    this->dataPtr->lastMarkersUpdateTime);
+
+  if (timeDiff.count() < this->dataPtr->markerLifetime)
+    return;
+
+  // Store simulation time
+  this->dataPtr->lastMarkersUpdateTime = _info.simTime;
+
+  // todo(anyone) Get the contacts of the links that don't have a
+  // contact sensor
+
+  // Get the contacts and publish them
+  // Since we are setting a lifetime for the markers, we get all the
+  // contacts instead of getting news and removed ones
+
+  // Variable for setting the markers id through the iteration
+  int markerID = 1;
+  _ecm.Each<components::ContactSensorData>(
+    [&](const Entity &,
+        const components::ContactSensorData *_contacts) -> bool
+    {
+      for (const auto &contact : _contacts->Data().contact())
+      {
+        // todo(anyone) add information about contact normal, depth
+        // and wrench
+        for (int i = 0; i < contact.position_size(); ++i)
+        {
+          // Skip dummy data set by physics
+          bool dataIsDummy = std::fabs(
+            contact.position(i).x() -
+            static_cast<float>(ignition::math::NAN_I)) < 0.001;
+          if (dataIsDummy)
+            return true;
+
+          // Set marker id, poses and request service
+          this->dataPtr->positionMarkerMsg.set_id(markerID);
+          ignition::msgs::Set(this->dataPtr->positionMarkerMsg.mutable_pose(),
+            ignition::math::Pose3d(contact.position(i).x(),
+              contact.position(i).y(), contact.position(i).z(),
+              0, 0, 0));
+
+          // Placeholder for the force value (see todo comment above)
```

Comment:
I'm a bit concerned about displaying force data that's not real. I think it's good that we have everything in place to display the force once that's available, but the end user doesn't know that this isn't the real data, so it may cause confusion. How about commenting this out for now?

Reply:
I don't understand, do you mean commenting out the `force` declaration and using a hardcoded value instead?

Reply:
Maybe I didn't understand the implementation then. I thought it was already hardcoded? Now it's always 1 scaled to the value the user inputs. It's not the real force, right? Also, the normal direction also doesn't seem to be correct?

Reply:
Yep, all questions are correct. I don't get what's the suggestion for avoiding confusion for the user.

Reply:
Oh gotcha, so I'd remove the green cylinders completely. We can leave all the code related to them commented out.

Reply:
Agree.